internal/commands/update_stemcell.go

```diff
@@ -69,44 +69,48 @@ func (update UpdateStemcell) Execute(args []string) error {
 	releaseSource := update.MultiReleaseSourceProvider(kilnfile, false)
 
 	for i, rel := range kilnfileLock.Releases {
-		update.Logger.Printf("Updating release %q with stemcell %s %s...", rel.Name, kilnfileLock.Stemcell.OS, trimmedInputVersion)
+		update.Logger.Printf("Updating release %s %s with stemcell %s %s...", rel.Name, rel.Version, kilnfileLock.Stemcell.OS, trimmedInputVersion)
 
 		spec, err := kilnfile.BOSHReleaseTarballSpecification(rel.Name)
 		if err != nil {
 			return err
 		}
+
 		spec.StemcellOS = kilnfileLock.Stemcell.OS
 		spec.StemcellVersion = trimmedInputVersion
 
-		var remote cargo.BOSHReleaseTarballLock
-		if update.Options.UpdateReleases {
-			remote, err = releaseSource.FindReleaseVersion(spec, true)
-		} else {
+		if !update.Options.UpdateReleases {
 			spec.Version = rel.Version
-			remote, err = releaseSource.GetMatchedRelease(spec)
 		}
 
+		remote, err := releaseSource.FindReleaseVersion(spec, true)
```

Comment:
`FindReleaseVersion` could return a newer version than desired since I don't believe all implementations use `spec.Version` if it is specified

In particular, I noticed the artifactory release source only  uses `spec.StemcellVersion` and the version constraint from the Kilnfile `spec.VersionConstraint` with FindReleaseVersion.

Before the --update-releases` option is added, this code path only used releaseSource.GetMatchedRelease

Reply:
Thanks @dlinsley. I was unsure how to proceed here. I assume you are thinking that this would work better?

```golang
if !update.Options.UpdateReleases {
  spec.Version = rel.Version
  remote, err = releaseSource.GetMatchedRelease(spec)
} else {
  remote, err = releaseSource.FindReleaseVersion(spec, true)
}
```

Reply:
Yes, I thought that would be more predictable.